src/SALib/analyze/discrepancy.py

```diff
@@ -0,0 +1,152 @@
+from typing import Dict
+
+import numpy as np
+from scipy.stats import qmc
+
+from SALib.analyze import common_args
+from SALib.util import read_param_file, ResultDict, _check_groups
+
+
+def analyze(
+    problem: Dict,
+    X: np.ndarray,
+    Y: np.ndarray,
+    method: str = "WD",
+    print_to_console: bool = False,
+    seed: int = None,
+):
+    """Discrepancy indices.
+
+    Parameters
+    ----------
+    problem : dict
+        The problem definition
+    X : numpy.ndarray
+        A NumPy array containing the model inputs
+    Y : numpy.ndarray
+        A NumPy array containing the model outputs
```

Comment:
I prefer `np.ndarray` for detailing the argument type, but is specifying the full name the accepted norm? I'll defer to what you say here.

The descriptor holds redundant info as well. I would state what the array holds directly instead, e.g., "An array of model inputs" (the type is already detailed)

Reply:
So we need the full name so that the cross links are working. This is because in `intersphinx_mapping` we use the key `numpy`. We could change this to `np` if you want (and update everywhere, I've seen that we use `numpy.array` sometimes and this should be `numpy.ndarray` at least.)

https://numpydoc.readthedocs.io/en/latest/format.html#parameters

Reply:
Okay, I prefer `np` (personal preference) but I will leave that as a separate task for the future.